content/posts/numpy/numpy-rng/index.md

```diff
@@ -0,0 +1,144 @@
+---
+title: "Good practices with NumPy random number generators"
+date: 2023-08-31T23:22:46+02:00
+draft: false
+description: "
+Good practices with NumPy random number generators
+"
+tags: ["tutorials", numpy, rng]
+displayInList: true
+author: ["Albert Thomas"]
+---
+
+Unless you are working on a problem where you can afford a true Random Number Generator (RNG), which is basically never for most of us, implementing something random means relying on a pseudo Random Number Generator. I want to share here what I have learned about good practices with pseudo RNGs and especially the ones available in [NumPy](https://numpy.org/).
```

Comment:
Well it seems that they exist according to a google search (e.g. https://en.wikipedia.org/wiki/Hardware_random_number_generator) but I agree that the term can be misleading and that they have their own issues. I just wanted to say here that PRNGs are not the only way to get random numbers but I will remove this.

Reply:
Yeah, it's a bit weird:

> While "classical" (non-quantum) phenomena are not truly random, an unpredictable physical system is usually acceptable as a source of randomness, so the qualifiers "true" and "physical" are used interchangeably.

In that case, one could consider /dev/urandom truly random as well, since it's connected to a piece of hardware.

Reply:
yes I learned something :)